src/generators/cplusplus/CplusplusFileGenerator.ts

```diff
@@ -0,0 +1,45 @@
+import { CplusplusGenerator, CplusplusRenderCompleteModelOptions } from '.';
+import { InputMetaModel, OutputModel } from '../../models';
+import * as path from 'path';
+import { AbstractFileGenerator } from '../AbstractFileGenerator';
+import { FileHelpers } from '../../helpers';
+
+export class CplusplusFileGenerator
+  extends CplusplusGenerator
+  implements AbstractFileGenerator<CplusplusRenderCompleteModelOptions>
+{
+  /**
+   * Generates all the models to an output directory each model with their own separate files.
+   *
+   * @param input
+   * @param outputDirectory where you want the models generated to
+   * @param options
+   */
+  public async generateToFiles(
+    input: Record<string, unknown> | InputMetaModel,
+    outputDirectory: string,
+    options?: CplusplusRenderCompleteModelOptions,
+    ensureFilesWritten = false
+  ): Promise<OutputModel[]> {
+    let generatedModels = await this.generateCompleteModels(
+      input,
+      options || {}
+    );
+    //Filter anything out that have not been successfully generated
+    generatedModels = generatedModels.filter((outputModel) => {
+      return outputModel.modelName !== '';
+    });
+    for (const outputModel of generatedModels) {
+      const filePath = path.resolve(
+        outputDirectory,
+        `${outputModel.modelName}.cpp`
```

Comment:
It would make more sense to make the generated files headers, with *.hpp extension.

Reply:
So we only generate header files, no `.cpp` files? 🤔

Reply:
Yes, we are basically creating an interface for data storage, and interfaces need to be shared between compilation units, that's what headers achieve.

Reply:
Okay, so we can change all the models to be located in header files instead?

Reply:
Alright, switched 👍